### PR TITLE
Initial draft for triaging

### PR DESCRIPTION
**Pull request by Sherwin-14**
This is the initial draft for the triaging document(#754). The pre-commit check is failing because of the use of format argument which seems like a necessity for rendering the doc, more on this [here](https://mkdocs-mermaid2.readthedocs.io/en/latest/superfences/#usage-for-the-material-theme).

----
📚 Documentation preview 📚: https://earthaccess--901.org.readthedocs.build/en/901/

### CODE REVIEW

**Comment by mfisher87** [comment]
Nice work! Will look more closely this week.

Looks like there's a tradeoff to be considered here to fix linting: https://github.com/pre-commit/pre-commit-hooks#check-yaml

**Comment by asteiker** [comment]
I'm wondering if some updates may be needed based on the recently closed #760 ? If it would be helpful for me to review and update accordingly, I'd be happy to help.

**Comment by mfisher87** [comment]
@asteiker that would be super helpful!!

**Comment by mfisher87** [comment]
@asteiker OK to assign this PR to you?

File: docs/contributing/triaging.md [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+```
+[label name](https://github.com/username/repository/labels/label-name)
+```
+
+### Example
+
+For example, to link to the "hackathon" label in the nsidc/earthaccess repository, you would use the following syntax:
+
+```
+[hackathon](https://github.com/nsidc/earthaccess/labels/hackathon)
```
[the remark this answers is not in the document]
Comment (mfisher87) [reply docs/contributing/triaging.md:59]: Should actually look like https://github.com/nsidc/earthaccess/labels/good%20first%20issue - GitHub automatically applies the text when you paste the URL :)

```
Should actually look like https://github.com/nsidc/earthaccess/labels/good%20first%20issue
```

File: docs/contributing/triaging.md [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+
+When referencing a label in a GitHub issue or discussion, it will be  useful to link to the label page to provide additional context and help other members to quickly understand the issue's category.
+
+### Syntax
+
+To link to a label in GitHub Markdown, use the following syntax:
+
+```
+[label name](https://github.com/username/repository/labels/label-name)
+```
```
Comment (mfisher87) [inline docs/contributing/triaging.md:52]:
````suggestion
To link to a label in GitHub Markdown, copy-and-paste the URL to the label by right-clicking [any label](https://github.com/nsidc/earthaccess/labels) and selecting "Copy Link". Then, paste that label in a GitHub issue, PR, discussion, or Markdown document:

```
https://github.com/username/repository/labels/label-name
```
````

File: docs/contributing/triaging.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (mfisher87) [reply docs/contributing/triaging.md:59]: After our example, let's include an image that shows what it should look like when done correctly!

File: docs/contributing/triaging.md [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+### Example
+
+For example, to link to the "hackathon" label in the nsidc/earthaccess repository, you would use the following syntax:
+
+```
+[hackathon](https://github.com/nsidc/earthaccess/labels/hackathon)
+```
+
+!!! tip
+    When linking to labels, it is a good practice to use the label name as the link text, as shown in the example above. This makes it clear to others what label is being referenced, and helps to avoid confusion.
```
Comment (mfisher87) [inline docs/contributing/triaging.md:63]:
```suggestion
```

This is a great tip, but if the label syntax is correct, the user shouldn't have to worry about that because GitHub handles the magic :)

File: docs/contributing/triaging.md [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+
+## Close Issue as Not Planned
+
+This section will cover the guidelines for when to use the "Close Issue as Not Planned" label, and how to handle issues that are not planned or feasible.
```
Comment (mfisher87) [inline docs/contributing/triaging.md:67]:
```suggestion
This section will cover the guidelines for when to use the "Close Issue as Not Planned" feature, and how to handle issues that are not planned or feasible.
```

File: docs/contributing/triaging.md [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+
+### When to Use This Label?
```
Comment (mfisher87) [inline docs/contributing/triaging.md:69]:
```suggestion
### When to "Close as not planned"?
```

File: docs/contributing/triaging.md [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+
+Use the "Close Issue as Not Planned" label when:
```
Comment (mfisher87) [inline docs/contributing/triaging.md:71]:
```suggestion
Close issues as "not planned" when:
```

File: docs/contributing/triaging.md [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+
+- An issue is not aligned with the project's goals or priorities.
+- An issue is not feasible to be addressed due to technical or resource constraints.
+- An issue is a duplicate of an existing issue that has already been addressed.
+
+When closing an issue as not planned, keep the following best practices in mind:
+
+- Provide a clear explanation as to why the issue is not planned or feasible.
+- Offer alternative solutions or workarounds, if possible.
+- Link to relevant documentation or resources, if applicable.
```
Comment (mfisher87) [inline docs/contributing/triaging.md:81]: :100: Amazing callout!

File: docs/contributing/triaging.md [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+## Discussions vs Issues
+
+This section would cover the guidelines for when to use discussions versus issues, and how to migrate between them.
+
+###  What are Discussions?
+
+Discussions are used for:
+
+- Brainstorming and idea generation.
+- Feature requests and feedback.
```
Comment (mfisher87) [inline docs/contributing/triaging.md:92]:
```suggestion
- Project feedback.
```

I wanted to differentiate this from "Requesting changes or improvements" in the "What are issues?" section :)

File: docs/contributing/triaging.md [hunk 9]
```diff
@@ -0,0 +1,149 @@
...
+
+###  What are Issues?
+
+Issues are used for:
+
+- Reporting bugs and errors.
+- Tracking progress on specific tasks or projects.
+- Requesting changes or improvements.
+
+### When to Migrate
```
Comment (mfisher87) [inline docs/contributing/triaging.md:103]:
```suggestion
### When to Migrate

Use your best judgement when migrating between issues and discussions. Sometimes it makes more sense to open a new issue or discussion instead of migrating, for example when there are many things being discussed, but we want to create an issue or task out of just one of those things.
```

File: docs/contributing/triaging.md [hunk 10]
```diff
@@ -0,0 +1,149 @@
...
+
+Migrate a discussion to an issue when:
+
+- A specific task is identified.
+- A bug or error is reported.
+- A change or improvement is requested.
+
+Migrate an issue to a discussion when:
+
+- The issue is a feature request or idea.
```
Comment (mfisher87) [inline docs/contributing/triaging.md:113]:
```suggestion
- The issue is a nebulous idea that needs to be workshopped before it can be implemented.
```

**Review by mfisher87 — CHANGES_REQUESTED** [review]
:raised_hands: Amazing work! Sorry about the number of suggestions here -- we re-organized labels in #760 to be more self-describing, and I think that helps make this document more clear!

**Comment by mfisher87** [comment]
@asteiker can you also take a look?

File: mkdocs.yml [hunk 11]
```diff
@@ -33,6 +33,7 @@ extra_css:
 
 plugins:
   - search
+  - mermaid2
```
Comment (mfisher87) [inline mkdocs.yml:36]: We need to install this plugin `mkdocs-mermaid2-plugin` in our docs dependencies to get the docs build working again:

https://github.com/nsidc/earthaccess/blob/91866ac18e806e2d53d6b68cd000fa372c3e010f/pyproject.toml#L90-L123

**Comment by mfisher87** [comment]
I'm not sure exactly what you mean here; are you thinking about PR reviews?

**Comment by mfisher87** [comment]
@Sherwin-14 I'd like to wait to review until the ReadTheDocs build is working again. Do you feel comfortable fixing it? Details here: https://github.com/nsidc/earthaccess/pull/901#discussion_r1912497324

File: mkdocs.yml [hunk 12]
```diff
@@ -67,6 +68,7 @@ nav:
       - "Maintainers Guide": "contributing/maintainers-guide.md"
       - "Code of Conduct": "contributing/code-of-conduct.md"
       - "Contributing naming convention": "contributing/naming-convention.md"
+      - "Triaging Doc": "contributing/triaging.md"
```
Comment (mfisher87) [inline mkdocs.yml:71]:
```suggestion
      - "Maintainers Guide": "contributing/maintainers-guide.md"
      - "Triaging Guide": "contributing/triaging.md"
      - "Code of Conduct": "contributing/code-of-conduct.md"
      - "Contributing naming convention": "contributing/naming-convention.md"

```

File: docs/contributing/triaging.md [hunk 13]
```diff
@@ -0,0 +1,141 @@
...
+- **impact: dependencies**: Use this label for issues concerning dependencies.
+- **impact: documentation**: Use this label for issues related to documentation.
+- **impact: automation**: Use this label for issues related to the CI/CD pipeline or automation
+
+### Needs labels
+
+- **needs: decision**: We're struggling to decide what to do and the decision committee needs to help.
+- **needs: feedback**:  Use this label for issues where feedback is requested from the team or our community.
+- **needs: help**: Use this label for issues where additional help or contributions are needed.
+
```
Comment (mfisher87) [inline docs/contributing/triaging.md:32]: I'd like to add back the "good first issue" label, perhaps alone in a "Special labels" section? That is one that triagers should be especially aware of!

File: docs/contributing/triaging.md [hunk 14]
```diff
@@ -0,0 +1,141 @@
...
+  classDef default font-size:32pt;
+  start{"`Followed
+  issue
+  template?`"}
+  start ==NO==> close1[Change
+  to draft and ask to follow template]
+  start == YES ==> dupe{Is duplicate?}
+  dupe == YES ==> close2[Close and point to duplicate]
+  dupe == NO ==> repro{Has proper reproduction?}
+  repro == NO ==> close3[Label: 'needs reproduction' bot will auto close if no update has been made in 3 days]
```
Comment (mfisher87) [inline docs/contributing/triaging.md:124]: >bot will auto close if no update has been made in 3 days

I think this isn't true. We have one automation for autoclosing issues, but I believe it's broken: https://github.com/nsidc/earthaccess/blob/main/.github/workflows/issue-manager.yml

We should be careful about fixing it: #895 & https://jacobtomlinson.dev/posts/2024/most-stale-bots-are-anti-user-and-anti-contributor-but-they-dont-have-to-be/

File: docs/contributing/triaging.md [hunk 15]
```diff
@@ -0,0 +1,141 @@
...
+  repro == YES ==> real{Is actually a bug?}
+  real == NO ==> intended{Is the intended behaviour?}
+  intended == YES ==> explain[Explain and close point to docs if needed]
+  intended == NO ==> open[Keep open for discussion Remove 'pending triage' label]
+  real == YES ==> real2["1. Remove 'pending triage' label 2. Add related feature label if applicable (e.g. 'feat: ssr') 3. Add priority and meta labels (see below)"]
```
Comment (mfisher87) [inline docs/contributing/triaging.md:129]: Renders as 
image

File: docs/contributing/triaging.md [hunk 16]
```diff
@@ -0,0 +1,141 @@
...
+  real2 ==> unusable{Does the bug make earthaccess unusable?}
+  unusable == YES ==> maj{Does the bug affect the majority of earthaccess users?}
+  maj == YES ==> p5[p5: urgent]
+  maj == NO ==> p4[p4: important]
+  unusable == NO ==> workarounds{Are there workarounds for the bug?}
+  workarounds == NO ==> p3[p3: minor bug]
+  workarounds == YES ==> p2[p2: edge case has workaround]
```
Comment (mfisher87) [inline docs/contributing/triaging.md:136]: We don't have priority levels yet to my knowledge. @asteiker what do you think?

**Review by mfisher87 — COMMENTED** [review]
In a rush, just leaving comments :)

File: docs/contributing/triaging.md (hunk 14 above)
Comment (mfisher87) [reply docs/contributing/triaging.md:124]: Ah, that makes sense, thanks for the context!

Comment (mfisher87) [reply docs/contributing/triaging.md:127]: I think getting @asteiker 's eyes on this is probably the best way forward. Will you be at next week's hack day? Maybe we can work together to finalize the flowchart.

File: docs/contributing/triaging.md [hunk 17]
```diff
@@ -0,0 +1,144 @@
...
+
+### Needs labels
+
+- **needs: decision**: We're struggling to decide what to do and the decision committee needs to help.
+- **needs: feedback**:  Use this label for issues where feedback is requested from the team or our community.
+- **needs: help**: Use this label for issues where additional help or contributions are needed.
+
+### Special labels
+
+- **good-first-issue**: Use this label for issues that are suitable for new contributors. These issues are designed to be approachable and not overly complex, making them an ideal starting point for those looking to contribute to the project.
```
Comment (mfisher87) [inline docs/contributing/triaging.md:35]:
```suggestion
- **good first issue**: Use this label for issues that are suitable for new contributors. These issues are designed to be approachable and not overly complex, making them an ideal starting point for those looking to contribute to the project.
```

File: docs/contributing/triaging.md (hunk 14 above)
Comment (asteiker) [reply docs/contributing/triaging.md:127]: @mfisher87 @Sherwin-14 Apologies for my radio silence on this recently! Yes I am happy to get back into this and work together at the next hackday. I'll review this in more detail in preparation.

Comment (mfisher87) [reply docs/contributing/triaging.md:127]: Thanks, Amy!

**Comment by asteiker** [comment]
@mfisher87 I'm sharing some notes from reviewing with @Sherwin-14 during our hackday:
- It looks like there is an erroneous "Unsupported markdown: list" box but I believe Sherwin needs some guidance from you on how to remove.
- There is a step to Remove "pending triage" label, but we don't have that label to begin with. I suggest we open a new Issue to add this label to all newly entered Issues. (side note: I'd love to contribute to this if it's not too technical. If you have any guidance I'd love to hear!) I wouldn't want this to hold up the PR but otherwise that step won't make a lot of sense before we have the "pending triage" label to begin with.
- Adding thoughts on priority levels in your comment above.
- I suggest we soften the language on issues that do not follow a template (the first "no" fork). I wouldn't want this to create a perceived barrier to contributing an issue if they are basically asked to redo what they wrote originally. I believe there may be cases where either a user does not understand the difference between issue templates or feel that their issue doesn't fully belong in one or the other. Could we instead say "Identify whether an existing template is appropriate and migrate the issue to that template on behalf of the issue reporter."

File: docs/contributing/triaging.md (hunk 16 above)
Comment (mfisher87) [reply docs/contributing/triaging.md:139]: I think that sounds reasonable!

**Comment by mfisher87** [comment]
1. I have a [comment above](https://github.com/nsidc/earthaccess/pull/901#discussion_r1917026827) for this; we just need to make the thing not-numbered. I believe Sherwin and I worked on this together in a hack week and made some progress. @Sherwin-14 is there a chance you have some un-pushed work?
2. Good idea. My original advice to Sherwin was to drop this step in the flowchart to keep it simple and aim to represent how we work now. But if we think we're ready to adopt that workflow, let's do it! I think it's not too technical :) To keep with our existing patterns, how about `Needs: triage`? I suggest implementing this in the issue templates. We already have this but it doesn't work because the label name in the template doesn't match any real labels: https://github.com/nsidc/earthaccess/blob/74c0ef50b43a05fc100c2245fcaf45b472dceb0b/.github/ISSUE_TEMPLATE/bug.yml?plain=1#L7
3. Responded above :) 
4. Yes! Great call. Perhaps "request needed information from reporter and update issue on behalf of reporter"?

If we're seeing users struggling to figure out which template to use, maybe we could do more to guide them in [our config](https://github.com/nsidc/earthaccess/blob/main/.github/ISSUE_TEMPLATE/config.yml)?

**Comment by mfisher87** [comment]
@Sherwin-14 how are you feeling about the next steps for this PR?

**Comment by Sherwin-14** [comment]
> @Sherwin-14 how are you feeling about the next steps for this PR?

I was waiting for the next hackday to start working on this again. I believe collaborating on this alongside @asteiker would be the ideal path forward. We can list all the areas where improvements are needed, in that way I would also get an idea about what steps I need to take moving forward. I had individual reviews from both of you, but I feel if all of us collaborate together on this coming hackday, that would make a lot easier for us to keep a track of suggestions and improvements.

**Comment by asteiker** [comment]
@Sherwin-14 Yes, let's finalize these open items together at the hackday. I would also like to start implementing the priority designation in the bug priority project view: https://github.com/orgs/nsidc/projects/10

**Comment by Sherwin-14** [comment]
> @Sherwin-14 Yes, let's finalize these open items together at the hackday. I would also like to start implementing the priority designation in the bug priority project view: https://github.com/orgs/nsidc/projects/10

Hey Amy! Could you summarize the improvements that we agreed upon on the last hackday? I do remember some of those but I guess you have a better idea about this.

**Comment by asteiker** [comment]
@Sherwin-14 Here is a summary of the changes remaining:

1. Update the erroneous "Unsupported markdown: list" box. This needs to be updated to be not numbered. There may be un-pushed work on this already from a previous hackday. 

2. Update language on issues that do not follow a template (the first "no" fork). Update to: "request needed information from reporter and update issue on behalf of reporter"

Note that I also just entered https://github.com/nsidc/earthaccess/issues/984 and can work this asap so that the pending triage label removal step is applicable.

**Comment by Sherwin-14** [comment]
> @Sherwin-14 Here is a summary of the changes remaining:
> 
>     1. Update the erroneous "Unsupported markdown: list" box. This needs to be updated to be not numbered. There may be un-pushed work on this already from a previous hackday.
> 
>     2. Update language on issues that do not follow a template (the first "no" fork). Update to: "request needed information from reporter and update issue on behalf of reporter"
> 
> 
> Note that I also just entered #984 and can work this asap so that the pending triage label removal step is applicable.

Thanks @asteiker! I have a question for you on the first point. Do you want me to remove the priority labels and just preserve the tags or do you want me to remove the tags alongside priority labels.

On removing priority labels and preserving the respective tags,  the workflow diagram would look something like this.

Screenshot From 2025-05-03 16-37-14

**Comment by mfisher87** [comment]
>Thanks @asteiker! I have a question regarding the first point. Do you want me to remove the priority labels and just preserve the tags or do you want me to remove the tags alongside priority labels.
>
> On removing priority labels and preserving the respective tags, the workflow diagram would look something like this.

Can you clarify what you mean by "tags" here? If you mean the workflow steps (rectangles) I'm all for removing them :)

**Comment by mfisher87** [comment]
:100:

I think we can remove  "Does it make earthaccess unusable" and everything underneath. Let's worry about prioritization of issues when we're ready :)

File: mkdocs.yml [hunk 18]
```diff
@@ -121,7 +124,12 @@ markdown_extensions:
       anchor_linenums: true
   - pymdownx.inlinehilite
   - pymdownx.snippets
-  - pymdownx.superfences
+  - pymdownx.superfences:
+      custom_fences:
+        - name: mermaid
+          class: mermaid
+          format: !!python/name:mermaid2.fence_mermaid_custom
```
Comment (mfisher87) [inline mkdocs.yml:131]: Pre-commit is complaining about this line, and I'm not sure why! Anything we can do about that?

**Review by mfisher87 — DISMISSED** [review]
Let's merge this once the pre-commit error is resolved. Thank you for your patience with this incredibly persistent task :sweat_smile:

File: mkdocs.yml (hunk 18 above)
Comment (mfisher87) [reply mkdocs.yml:131]: Looks like we can add `--unsafe` arg to pre-commit to get by: https://github.com/pre-commit/pre-commit-hooks#check-yaml

Comment (Sherwin-14) [reply mkdocs.yml:131]: > Pre-commit is complaining about this line, and I'm not sure why! Anything we can do about that?

I don't know tbh. It seems like this line is important to render the flowchart, more on this [here](https://mkdocs-mermaid2.readthedocs.io/en/latest/superfences/#usage-for-the-material-theme).

Comment (mfisher87) [reply mkdocs.yml:131]: Yeah, I don't think we should change the mkdocs config. We should change the pre-commit config! See the link above :)

**Comment by Sherwin-14** [comment]
`pre-commit` now fails because of a uv related error.

**Comment by mfisher87** [comment]
Nice, thanks for the fix! The `uv` failure has been happening for a long while, no worries! It needs a separate fix.

**Comment by mfisher87** [comment]
This PR is docs-only so ignoring the integration test failures

**Review by mfisher87 — APPROVED** [review]

**Comment by mfisher87** [comment]
Thanks, @Sherwin-14  :tada:
